src/relay/op/tensor/reduce.cc

```diff
@@ -193,15 +194,19 @@ Array<te::Tensor> ReduceCompute(const Attrs& attrs, const Array<te::Tensor>& inp
 /*!
  * \brief ReduceShapeImpl get the outshape for the reduction operator
  * \param in_shape Shape of input data.
- * \param param ReduceAttrs details.
+ * \param inaxis The input axis vector.
+ * \param keepdims If this is set to true, the axes which are reduced are
+ * left in the result as dimensions with size one. This enables the result
+ * to broadcast correctly against the input array.
+ * \param exclude Whether 'axis' input given is the excluded axis.
  * \param reporter The reporter to report solution to.
  * \return oshape Output shape inferred.
  */
 inline std::vector<IndexExpr> ReduceShapeImpl(const std::vector<IndexExpr>& in_shape,
-                                              const ReduceAttrs* param,
-                                              const TypeReporter& reporter) {
```

Comment:
Let's make `param` here a template, so that `ReduceShapeImpl` can take both `ReduceAttrs` and `VarianceAttrs` (since they have the same attributes)